pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -142,31 +142,65 @@ func testHTTPStats(t *testing.T, aggregateByStatusCode bool) {
 }
 
 func TestHTTPSViaLibraryIntegration(t *testing.T) {
-	if !httpSupported(t) {
-		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
-	}
 	if !httpsSupported(t) {
 		t.Skip("HTTPS feature not available/supported for this setup")
 	}
 
+	buildPrefetchFileBin(t)
+
 	tlsLibs := []*regexp.Regexp{
 		regexp.MustCompile(`/[^\ ]+libssl.so[^\ ]*`),
 		regexp.MustCompile(`/[^\ ]+libgnutls.so[^\ ]*`),
 	}
 	tests := []struct {
-		name     string
-		fetchCmd []string
+		name         string
+		fetchCmd     []string
+		prefetchLibs []string
 	}{
-		{name: "wget", fetchCmd: []string{"wget", "--no-check-certificate", "-O/dev/null"}},
-		{name: "curl", fetchCmd: []string{"curl", "--http1.1", "-k", "-o/dev/null"}},
+		{
+			name:     "wget",
+			fetchCmd: []string{"wget", "--no-check-certificate", "-O/dev/null"},
+		},
+		{
+			name:     "curl",
+			fetchCmd: []string{"curl", "--http1.1", "-k", "-o/dev/null"},
+		},
+	}
+
+	for _, test := range tests {
+		fetch, err := exec.LookPath(test.fetchCmd[0])
+		if err != nil {
+			t.Skipf("%s not found; skipping test.", test.fetchCmd)
+		}
+		ldd, err := exec.LookPath("ldd")
+		if err != nil {
+			t.Skip("ldd not found; skipping test.")
+		}
+		linked, _ := exec.Command(ldd, fetch).Output()
+
+		for _, lib := range tlsLibs {
+			libSSLPath := lib.FindString(string(linked))
+			if _, err := os.Stat(libSSLPath); err == nil {
+				test.prefetchLibs = append(test.prefetchLibs, libSSLPath)
+			}
+		}
+		if len(test.prefetchLibs) == 0 {
+			t.Fatalf("%s not linked with any of these libs %v", test.name, tlsLibs)
```

Comment:
The downside of this approach is that the failure won't be attributed to `test.name` in Go's eyes, only via the log message.

Reply:
I'll move that failure line to the actual test

Reply:
If you move the `tests` loop to the outside and have `keepAlives` be the inner loop, then you can do this logic only once, while still retaining the failure association.

Reply:
that was my original intention, but we're setting the server (that's the heavier operation) only once per configuration (affected by keep-alive flag)

Reply:
Got it. That makes sense.